x-pack/filebeat/input/gcs/job/job.go

```diff
@@ -0,0 +1,179 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+// +build !aix
+
+package job
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/storage"
+
+	cursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/gcs/state"
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/gcs/types"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+type Job interface {
+	Do(ctx context.Context, id string) error
+	Name() string
+	Timestamp() *time.Time
```

Comment:
Why does this return a pointer?

Reply:
The pointer is not necessary, will remove it.